oak/server/enclave_server.h

```diff
@@ -141,38 +140,24 @@ class EnclaveServer final : public asylo::TrustedApplication {
     }
   }
 
-  void RequestNextCall() {
-    // The stream will delete itself when it finishes.
-    auto stream = std::make_shared<grpc_server::GrpcStream>(node_);
-    generic_service_.RequestCall(&stream->server_context(), &stream->server_reader_writer(),
-                                 completion_queue_.get(), completion_queue_.get(),
-                                 new grpc_server::StreamCreationEventHandler(stream));
-  }
-
   // Consumes gRPC events from the completion queue in an infinite loop.
   void CompletionQueueLoop() {
     LOG(INFO) << "Starting gRPC completion queue loop";
+    // The stream object will delete itself when finished with the request,
+    // after creating a new stream object for the next request.
+    auto *stream = new GrpcStream(&module_service_, completion_queue_.get(), node_.get());
+    stream->RequestNext();
     while (true) {
-      RequestNextCall();
-
-      grpc_server::BaseGrpcEventHandler *eventHandler = nullptr;
-      bool ok = false;
-      if (!completion_queue_->Next(reinterpret_cast<void **>(&eventHandler), &ok)) {
+      bool ok;
+      void *tag;
+      // TODO: Handle streaming requests?
```

Comment:
Probably not for the foreseeable future, since we currently serialise invocations anyways, so a long-lived streaming request would prevent any other request from being processed. Anyway feel free to leave the TODO around :)

Reply:
If we don't need to support streaming RPCs then this can be further simplified using GenericServerAsyncResponseWriter instead of GenericServerAsyncReaderWriter.
That would reduce the flow to Request -> Process -> Finish.